brownie/network/contract.py

```diff
@@ -127,6 +130,153 @@ def decode_input(self, calldata: Union[str, bytes]) -> Tuple[str, Any]:
 
         return function_sig, input_args
 
+    def get_verification_info(self) -> Dict:
+        """
+        Return a dict with flattened source code for this contract
+        and further information needed for verification
+        """
+        language = self._build["language"]
+        if language == "Vyper":
+            raise TypeError(
+                "Etherscan does not support API verification of source code "
+                "for vyper contracts. You need to verify the source manually"
+            )
+        elif language == "Solidity":
+            flattened_source = ""
+            has_abiencoder = False
+            abiencoder_re = r"(^|\r|\n|\r\n)\s*pragma experimental ABIEncoderV2;"
+
+            dependency_tree: Dict = defaultdict(set)
+            dependency_tree["__root_node__"] = set(self._build["dependencies"])
+            code_fragments: Dict = {}
+
+            import_aliases: Dict = defaultdict(list)
+            used_offsets: Dict = defaultdict(list)  # aggregate all used offsets per source file
+
+            for name in self._build["dependencies"]:
+                build_json = self._project._build.get(name)
+                if "dependencies" in build_json:
+                    dependency_tree[name].update(build_json["dependencies"])
+
+                # Find import directive nodes with symbolAliases
+                for node in build_json["ast"]["nodes"]:
+                    if node["nodeType"] == "ImportDirective" and len(node["symbolAliases"]) > 0:
+                        for symbol_alias in node["symbolAliases"]:
+                            if symbol_alias["local"] is not None:
+                                import_aliases[symbol_alias["foreign"]["name"]].append(
+                                    symbol_alias["local"]
+                                )
+
+                offset = slice(*build_json["offset"])
+                used_offsets[build_json["source"]].append(tuple(build_json["offset"]))
+                source = self._slice_source(build_json["source"], offset)
+                code_fragments[name] = [source, name]
+                if not has_abiencoder:
+                    has_abiencoder = (
+                        len(re.findall(abiencoder_re, build_json["source"][: offset.start])) > 0
+                    )
+
+            for name in toposort_flatten(dependency_tree):
+                if name == "__root_node__":
+                    continue
+                source, part_name = code_fragments[name]
+                # Create a duplicate entry for each import alias
+                part_name_short = part_name
+                if "/" in part_name:
+                    part_name_short = part_name.split("/")[-1]
```

Comment:
We know for sure this is OK on windows, yes? Solidity always uses unix-style paths?

Reply:
Yes, this is taken from the build jsons, the path is always with /.